src/lib/util/util.ts

```diff
@@ -215,42 +215,49 @@ export async function fetchAllLeaderboardEntries(client: Client, results: readon
 	return payload;
 }
 
-export async function fetch(url: URL | string, type: 'json'): Promise<object>;
-export async function fetch(url: URL | string, options: RequestInit, type: 'json'): Promise<object>;
-export async function fetch(url: URL | string, type: 'buffer'): Promise<Buffer>;
-export async function fetch(url: URL | string, options: RequestInit, type: 'buffer'): Promise<Buffer>;
-export async function fetch(url: URL | string, type: 'text'): Promise<string>;
-export async function fetch(url: URL | string, options: RequestInit, type: 'text'): Promise<string>;
-export async function fetch(url: URL | string, type: 'result'): Promise<Response>;
-export async function fetch(url: URL | string, options: RequestInit, type: 'result'): Promise<Response>;
-export async function fetch(url: URL | string, options: RequestInit, type: 'result' | 'json' | 'buffer' | 'text'): Promise<Response | Buffer | string | object>;
-export async function fetch(url: URL | string, options: RequestInit | 'result' | 'json' | 'buffer' | 'text', type?: 'result' | 'json' | 'buffer' | 'text') {
+export const enum FetchResultTypes {
+	JSON,
+	Buffer,
+	Text,
+	Result
+}
+
+export async function fetch(url: URL | string, type: FetchResultTypes.JSON): Promise<unknown>;
+export async function fetch(url: URL | string, options: RequestInit, type: FetchResultTypes.JSON): Promise<unknown>;
+export async function fetch(url: URL | string, type: FetchResultTypes.Buffer): Promise<Buffer>;
+export async function fetch(url: URL | string, options: RequestInit, type: FetchResultTypes.Buffer): Promise<Buffer>;
+export async function fetch(url: URL | string, type: FetchResultTypes.Text): Promise<string>;
+export async function fetch(url: URL | string, options: RequestInit, type: FetchResultTypes.Text): Promise<string>;
+export async function fetch(url: URL | string, type: FetchResultTypes.Result): Promise<Response>;
+export async function fetch(url: URL | string, options: RequestInit, type: FetchResultTypes.Result): Promise<Response>;
+export async function fetch(url: URL | string, options: RequestInit, type: FetchResultTypes): Promise<Response | Buffer | string | unknown>;
+export async function fetch(url: URL | string, options: RequestInit | FetchResultTypes, type?: FetchResultTypes) {
 	if (typeof options === 'undefined') {
 		options = {};
-		type = 'json';
+		type = FetchResultTypes.JSON;
 	} else if (typeof options === 'string') {
```

Comment:
By default, enum values are numbers:
```suggestion
	} else if (typeof options === 'number') {
```